packages/framework/presence/src/presence.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ConnectedClientId } from "./baseTypes.js";
+import type { NotificationsManager } from "./notificationsManager.js";
+import type {
+	PresenceStates,
+	PresenceWorkspaceAddress,
+	PresenceStatesSchema,
+} from "./types.js";
+
+import type { ISubscribable } from "@fluid-experimental/presence/internal/events";
+import type { InternalTypes } from "@fluid-experimental/presence/internal/exposedInternalTypes";
+
+/**
+ * A client within a Fluid session (period of container connectivity to service).
+ *
+ * @sealed
+ * @alpha
+ */
+export interface ISessionClient {
+	/**
+	 * Get current client connection id.
+	 *
+	 * @returns Current client connection id.
+	 *
+	 * @remarks
+	 * Connection id will change on reconnection.
+	 */
+	currentClientId(): ConnectedClientId;
+}
+
+/**
+ * @sealed
+ * @alpha
+ */
+export interface PresenceEvents {
+	/**
+	 * Raised when new client joins session.
+	 *
+	 * @eventProperty
+	 */
+	attendeeJoined: (attendee: ISessionClient) => void;
+
+	/**
+	 * Raised when client appears disconnected from session.
+	 *
+	 * @eventProperty
+	 */
+	attendeeDisconnected: (attendee: ISessionClient) => void;
+
+	/**
+	 * Raised when a workspace is activated within the session.
+	 */
+	workspaceActivated: (
+		workspaceAddress: PresenceWorkspaceAddress,
+		type: "States" | "Notifications" | "Unknown",
```

Comment:
I think this event means that either getStates or getNotifications (depending on the `type`) now has a new member?

If so, maybe consider having two separate events rather than a type param, allowing you to just name them "statesAdded" and "notificationsAdded" to better mirror the retrieval APIs?  Or alternatively, merge getStates+getNotifications into getWorkspaces?  It seems asymmetrical to have a unified event but split retrieval methods.

Reply:
See above notes - actually means there is session activity.
I wasn't sure if this should bother firing for when one is added locally. It probably doesn't make a lot of sense to do that. I expect most local cases will not share workspaces and if they do they won't be passive.